src/components/DateRangePicker.js

```diff
@@ -313,42 +312,51 @@ class DateRangePicker extends React.Component {
       ...this.props.focusTrapProps
     };
 
+    const buttonProps = {
+      'aria-controls': 'calendarMenu',
+      'aria-haspopup': true,
+      'aria-label': `${placeholderText}${
+        this.props.selectedStartDate && this.props.selectedEndDate
+          ? `, ${selectedStartDateFromPropsAsMoment.format('MMMM Do, YYYY')} to ${selectedEndDateFromPropsAsMoment.format('MMMM Do, YYYY')} currently selected`
+          : ''
+      }`,
+      onClick: this._toggleSelectionPane
+    }
+
     return (
       <div className='mx-date-range-picker' style={styles.component}>
-        <button
-          aria-controls='calendarMenu'
-          aria-haspopup={true}
-          aria-label={`${placeholderText}${
-            this.props.selectedStartDate && this.props.selectedEndDate
-              ? `, ${selectedStartDateFromPropsAsMoment.format('MMMM Do, YYYY')} to ${selectedEndDateFromPropsAsMoment.format('MMMM Do, YYYY')} currently selected`
-              : ''
-          }`}
-          onClick={this._toggleSelectionPane}
-          ref={this.props.elementRef}
-          style={styles.selectedDateButton}
-        >
-          {shouldShowCalendarIcon ? (
-            <Icon
-              size={20}
-              style={styles.selectedDateIcon}
-              type='calendar'
-            />
-          ) : null}
-          <div className='mx-date-range-picker-selected-date-text' style={styles.selectedDateText}>
-            {this.props.selectedStartDate && this.props.selectedEndDate ? (
-              <div>
-                <span>{selectedStartDateFromPropsAsMoment.format(this._getDateFormat(isLargeOrMediumWindowSize))}</span>
-                <span> - </span>
-                <span>{selectedEndDateFromPropsAsMoment.format(this._getDateFormat(isLargeOrMediumWindowSize))}</span>
-              </div>
-            ) : placeholderText}
-          </div>
-          <Icon
-            size={20}
-            style={styles.selectedDateCaret}
-            type={this.state.showSelectionPane ? 'caret-up' : 'caret-down'}
-          />
-        </button>
+        {
+          children && typeof children === 'function' ? this.props.children(buttonProps) : 
```

Comment:
This is the main change here. If children and children are a function then we call it and pass the button Props that would have been spread across the normal button.

Reply:
I think it is a bit odd that you are using `children` and `this.props.children` here. It reads like they could be different things but they are the same thing. I would prefer that you use one or the other.

Reply:
Looks good and well documented